support/controlplane-component/job.go

```diff
@@ -48,32 +48,42 @@ func (c *jobProvider) Replicas(object *batchv1.Job) *int32 {
 
 // IsAvailable implements WorkloadProvider.
 func (c *jobProvider) IsAvailable(job *batchv1.Job) (status metav1.ConditionStatus, reason string, message string) {
-	complete := util.FindJobCondition(job, batchv1.JobComplete)
-	if complete != nil {
-		return metav1.ConditionStatus(complete.Status), complete.Reason, complete.Message
-	}
-	failed := util.FindJobCondition(job, batchv1.JobFailed)
-	if failed != nil {
-		return metav1.ConditionFalse, failed.Reason, failed.Message
-	}
 	if job.Status.Active > 0 {
 		return metav1.ConditionTrue, "JobActive", "Job is still running"
 	}
-	return metav1.ConditionFalse, "Unknown", "Job status unknown"
+	return JobCompletionStatus(job)
 }
 
 // IsReady implements WorkloadProvider.
 func (c *jobProvider) IsReady(job *batchv1.Job) (status metav1.ConditionStatus, reason string, message string) {
+	if job.Status.Active > 0 {
+		return metav1.ConditionFalse, "JobActive", "Job is still running"
+	}
+	return JobCompletionStatus(job)
+}
+
+// JobCompletionStatus checks the status of a job and returns the appropriate condition status, reason, and message.
+// It checks if the job is complete or failed and returns the corresponding status.
+// If the job is neither complete nor failed, it returns unknown status.
+func JobCompletionStatus(job *batchv1.Job) (status metav1.ConditionStatus, reason string, message string) {
 	complete := util.FindJobCondition(job, batchv1.JobComplete)
-	if complete != nil {
-		return metav1.ConditionStatus(complete.Status), complete.Reason, complete.Message
+	if complete != nil && complete.Status == corev1.ConditionTrue {
+		return metav1.ConditionTrue, "JobComplete", "Job completed successfully"
```

Comment:
why not just do the same logic as the `failed` case i.e. use the Reason/Message if set, else default to hardcoded reason/message?

Reply:
There wouldn't be anything interesting in the message if job completed successfully, figured we could save a few lines of code XD. 
can change it if you think that's better.